pkg/controller/clusterapi.go

```diff
@@ -61,3 +61,20 @@ func GetKubeadmControlPlane(ctx context.Context, client client.Client, cluster *
 	}
 	return kubeadmControlPlane, nil
 }
+
+// GetMachineDeployment reads a cluster-api MachineDeployment for an eks-a cluster using a kube client.
+// If the MachineDeployment is not found, the method returns (nil, nil).
+func GetMachineDeployment(ctx context.Context, client client.Client, cluster *anywherev1.Cluster, machineDeploymentName string) (*clusterv1.MachineDeployment, error) {
```

Comment:
nit: remove unused param
```suggestion
func GetMachineDeployment(ctx context.Context, client client.Client, machineDeploymentName string) (*clusterv1.MachineDeployment, error) {
```